lib/httparty/request.rb

```diff
@@ -89,6 +90,19 @@ def parser
       options[:parser]
     end
 
+    def URI uri
```

Comment:
`httparty` uses [URI's kernel method](https://github.com/ruby/ruby/blob/trunk/lib/uri/common.rb#L709) which is tightly coupled to Ruby `URI.parse` method. I thought the smallest discrete change that would help to introduce URI adaptor was to hide it from the `httparty` module. On a third commit I've inlined the method to make that more clear.